### ISSUE TITLE
Add bump_version

### ISSUE THREAD

**Issue by luisaforozco** [MALES-project/SpeckleCn2Profiler#28]
Let's use `bump_my_version` and setup the CI/CD to automate the bumping of the version every time we make a release.
Right now in `setup.py` there is: 
```python
version='{{VERSION_PLACEHOLDER}}',
```
That is gotten from the action `pypi_release.yaml` as `steps.tag.outputs.TAG_NAME`.

**Comment by luisaforozco** [comment]
The reason for using python 3.11.1 instead of python 3.11 is because of other dependencies like lie_learn?

**Comment by SCiarella** [comment]
The bump action has been fixed. 
Still, the [publication does not work](https://github.com/MALES-project/SpeckleCn2Profiler/actions/runs/8558315416/job/23452669673) due to the [outdated version](https://github.com/AMLab-Amsterdam/lie_learn/issues/27) of `lie_learn` on pypi.

**Comment by luisaforozco** [comment]
What about using python 3.10 for the [publish action](https://github.com/MALES-project/SpeckleCn2Profiler/blob/0c7ed0bed8f2660309c285e2c167d047523af8d6/.github/workflows/publish.yaml#L17)?

**Comment by SCiarella** [comment]
The problem is this
```
ERROR    HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/        
         Can't have direct dependency: lie-learn@                               
         git+https://github.com/AMLab-Amsterdam/lie_learn. See                  
         https://packaging.python.org/specifications/core-metadata for more     
         information.  
```
which is caused by requiring `lie-learn@ git+https://github.com/AMLab-Amsterdam/lie_learn` (pypi does not like git versions). 
If I simply require the pypi version `lie_learn`, then `speckcn2` can be uploaded on pypi, but it would not build.

**Comment by v1kko** [comment]
the dependency problem is fixed in #35
